### DISCUSSION TITLE
Question: would it be possible to share files on a ZeroTier virtual lan or is it technically not possible ?

### DISCUSSION THREAD

**Discussion by Virgus** [localsend/localsend#2772] [Discussion (English, International)] [answered]
This would be great to transfer files when you cannot connect directly to wlan but you can access it via zerotier over LTE connection, but I'm not sure if it's doable. Thanks.

**Comment by zillydev** [comment]
It is possible

**Comment by Virgus** [comment]
Thanks @zillydev for the good news.

I just tried activating zt on a new ZT network ID both on my pc and smartphone but I suppose I have to change the multicast IP to the one that zerotier uses.

...after reading [this](https://discuss.zerotier.com/t/multicast-problem/4884/3) I suppose I should test a newer version on Windows 10 (I'm still on version 1.6.6 , last version available on Win 7). What's the version you've successfully tested the ZT use case ?

I made a second test: I have an sbc (an Orange Pi) with the latest ZT docker image and I set my lan bridge according to [this link](https://4x5mg.net/2020/08/23/accessing-lan-over-zerotier/) with this specific network ID. So I turned off the ZT client on the pc and I can still ping my lan devices from my android smartphones. In this second case I tried adding multicast routing 224.0.0.0/24 -> 172.22.22.22 (MySBC_ZT_IP) in the "Managed routes" web GUI but it still didn't work.

It would be the best choice to use LocalSend's default settings and set ZT to accept them, so to avoid having to change settings when switching from LAN to ZT or viceversa.

I'm not sure if what I've done with Managed route's rule is correct and if I should (and how to) enable the multicast routing in the SBC host, or in the docker container, or in both...

Unfortunately I found a few other posts on the net on the topic, with many questions and not so many useful replies...
If you have any hint/advise to share it would be greatly appreciated.

Cheers,
V.

**Comment by Virgus** [comment, marked as answer]
JFYI, after more than a year I finally made it !
...thanks to a lot of reading and some guidance from ChatGPT.

My ZT setup consists in an SBC (an OrangePI) with the ZT client and IPtables set so that I can access the LAN hosts from ZT's virtual LAN. You can find instructions about this [here ](https://docs.zerotier.com/bridging/) and [here](https://docs.zerotier.com/route-between-phys-and-virt).

The extra configs that made me able to detect the LocalSend LAN hosts from my smartphone ( connected to ZT bridged network ) were set by the following commands:

```
#Allow Multicast
sudo iptables -A INPUT -d 224.0.0.0/4 -j ACCEPT

#Allow Multicast Discovery (UDP) on port 5353 (mDNS) to discover devices.
sudo iptables -A INPUT -p udp -m udp --dport 5353 -j ACCEPT
sudo iptables -A INPUT -p udp -m udp --sport 5353 -j ACCEPT

#Application Port 12345 for both TCP and UDP communications:
sudo iptables -A INPUT -p tcp --dport 12345 -j ACCEPT
sudo iptables -A INPUT -p udp --dport 12345 -j ACCEPT

#Broadcast Communication (UDP) for discovery traffic.
sudo iptables -A INPUT -p udp -m udp -d 255.255.255.255 --dport 12345 -j ACCEPT

#Forwarding Multicast need to traverse the ZeroTier network:
sudo iptables -A FORWARD -p udp --dport 5353 -j ACCEPT
sudo iptables -A FORWARD -p udp --dport 12345 -j ACCEPT

#Save iptables rules
sudo iptables-save > /etc/iptables/rules.v4
```
